features/FEATURE_BLE/ble/gap/AdvertisingParameters.h

```diff
@@ -162,6 +165,9 @@ class AdvertisingParameters {
     /**
      * Update the advertising type.
      *
+     * @note If legacy PDU is not used then you cannot use
+     * CONNECTABLE_UNDIRECTED nor CONNECTABLE_DIRECTED.
```

Comment:
Question, should we tie setting the advertising type and if legacy PDU is used or not ? 
I mean: `setType(advertising_type_t type, bool legacy)` ? 
Then it would make sense to assert if the type provided is not compatible with the non legacy PDUs.
That question also applies to the constructor.

Reply:
I don't see a difference between asserting in that new call or just asserting in the current call - the assert will exaplin the reason either way

Reply:
I take it back, I can see a difference - it's easier to set withour triggering the assert if there's call that lets you set the correct combinatin since the default value starts off as connectable undirected

Reply:
added function and checks